test/e2e/proxy_test.go

```diff
@@ -0,0 +1,67 @@
+package e2e
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/console-operator/test/e2e/framework"
+)
+
+func setupProxyTest(t *testing.T) (*framework.ClientSet, *operatorsv1.Console) {
+	clientSet, operatorConfig := framework.StandardSetup(t)
+	return clientSet, operatorConfig
+}
+
+func cleanupProxyTest(t *testing.T, clientSet *framework.ClientSet) {
+	defer framework.ResetClusterProxyConfig(clientSet)
+	framework.StandardCleanup(t, clientSet)
+}
+
+func TestProxy(t *testing.T) {
+	client, _ := setupProxyTest(t)
+	defer cleanupProxyTest(t, client)
+
+	clusterProxyConfig := configv1.ProxySpec{
+		NoProxy:    "clusternoproxy.example.com",
+		HTTPProxy:  "http://clusterhttpproxy.example.com",
+		HTTPSProxy: "https://clusterhttpsproxy.example.com",
+	}
+
+	proxyEnvVars := map[string][]corev1.EnvVar{
+		"emtpyVars": {
+			{Name: "NO_PROXY", Value: "", ValueFrom: nil},
+			{Name: "HTTP_PROXY", Value: "", ValueFrom: nil},
+			{Name: "HTTPS_PROXY", Value: "", ValueFrom: nil},
+		},
+		"clusterVars": {
+			{Name: "NO_PROXY", Value: clusterProxyConfig.NoProxy, ValueFrom: nil},
+			{Name: "HTTP_PROXY", Value: clusterProxyConfig.HTTPProxy, ValueFrom: nil},
+			{Name: "HTTPS_PROXY", Value: clusterProxyConfig.HTTPSProxy, ValueFrom: nil},
+		},
+	}
+
+	consoleDeployment, err := framework.GetConsoleDeployment(client)
+	if err != nil {
```

Comment:
So can this run without the Proxy CR?  This is a sufficient test?

Reply:
I could see on our end that passing the envs to the deployment is sufficient.

But do we then need a test in `console` validating that it responds through the proxy?

Reply:
We probably should. Lets talk about that :)